src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -263,5 +297,24 @@ private async Task<IEnumerable<CodeAction>> GetRazorCodeActionsAsync(RazorCodeAc
 
             return codeActions;
         }
+
+        private static HashSet<string> GetAllSupportedCodeActionNames()
+        {
+            var supportedCodeActionNames = new HashSet<string>();
+
+            var refactoringProviderNames = typeof(RazorPredefinedCodeRefactoringProviderNames)
```

Comment:
lol don't love this but I totally understand why you did it.

Reply:
Yeah, would've ideally avoided reflection but the alternative would've required manually updating every time based on the changes in Roslyn's Razor External Access, which is more trouble than it's worth.